src/features/Send/useCases/ListAmountsToSend/AddToken/SelectTokenFromListScreen.tsx

```diff
@@ -65,28 +115,80 @@ export const SelectTokenFromListScreen = () => {
           </Boundary>
         )}
         bounces={false}
-        contentContainerStyle={{paddingHorizontal: 16}}
+        contentContainerStyle={styles.list}
         keyExtractor={(_, index) => index.toString()}
         testID="assetsList"
         estimatedItemSize={78}
-        ListEmptyComponent={isSearchResultEmpty ? <EmptySearchResult /> : undefined}
+        ListEmptyComponent={assetSearchTerm.length > 0 && filteredTokenInfos.length === 0 ? <NoAssets /> : undefined}
       />
-
-      <View style={styles.counter}>
-        <Text style={styles.counterText1}>{strings.counter1(sortedTokenInfos.length)}</Text>
-
-        <Text style={styles.counterText2}>{` ${strings.counter2}`}</Text>
-      </View>
     </View>
   )
 }
 
+const useFilteredTokenInfos = ({activeTab}: {activeTab: Tabs}) => {
+  const wallet = useSelectedWallet()
+  const {nfts} = useNfts(wallet)
+  const {search: assetSearchTerm, visible} = useSearch()
+  const balances = useBalances(wallet)
+
+  const tokenInfos = useTokenInfos({
+    wallet,
+    tokenIds: Amounts.toArray(balances).map(({tokenId}) => tokenId),
+  })
+
+  const searchFilteredTokens = filterTokenInfos(assetSearchTerm, tokenInfos)
```

Comment:
I'd change filterTokenInfo function, to be just the `filter` function, and receive only 1 TokenInfo, this way we would be able to combine/reduce/map/filter/some, leverage all this functions, it could even be a curried function, that returns the filter function based on the rule applied -> `(desiredFilter/searchTerm) => (TokenInfo) => ...`

Reply:
it will become composable